File: samples/java/06cat/cat.java
import java.io.BufferedReader;
import java.io.FileNotFoundException;
import java.io.FileReader;
import java.io.IOException;
import java.io.InputStreamReader;

public class cat {

    public static void main(String[] args) {
        if (args.length > 1) {
            System.err.println("Error: Usage 'java cat [<file>]'");
            System.exit(1);
        }
        BufferedReader br = null;
        if (args.length == 1) {
            try { 
                br = new BufferedReader(new FileReader(args[0]));
            } catch (FileNotFoundException err) {
                System.err.printf("Error: File '%s' not found.\n", args[0]);
                System.exit(2);
            }
        } else {
            br = new BufferedReader(new InputStreamReader(System.in));
        }
        try {
            String line = br.readLine();
            while (line != null) {
                System.out.println(line);
                line = br.readLine();
            }
        } catch (IOException err) {
            err.printStackTrace();
        } finally {
            try {
                if (args.length == 1) {
                    br.close();
                }
            } catch (IOException err2) {
                System.err.printf("Failed to close %s\n", args[0]);
            }
        }
    }
}

File: samples/kotlin/06cat/cat.kt
import java.io.File;

fun main(args: Array<String>) {
    if (args.size > 1) {
        System.err.println("Error: Usage `kotlin cat [<file>]`");
        System.exit(1);
    }
    if (args.size == 1) {
        File(args[0]).forEachLine { line -> println(line) };
    } else {
        var line: String? = readLine();
        while (line != null) {
            println(line);
            line = readLine();
        }
    }
}
